### PR TITLE
CA-1143: update workbench-libs to adopt new bouncycastle version

### PR DESCRIPTION
**Pull request by MatthewBemis**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ object Dependencies {
   val workbenchUtil2V   = "0.1-74c9fc2"
   val workbenchModelV  = "0.14-74c9fc2"
   val workbenchGoogleV = "0.21-51d7fff"
-  val workbenchGoogle2V = "0.18-74c9fc2"
+  val workbenchGoogle2V = "0.21-712e75f9-SNAP"
```
Comment (MatthewBemis) [inline project/Dependencies.scala]: will update to non-snap version when it's published

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/config/AppConfig.scala [hunk 2]
```diff
@@ -96,7 +96,7 @@ object AppConfig {
     )
   }
 
-  val jsonFactory = JacksonFactory.getDefaultInstance
+  val jsonFactory = GsonFactory.getDefaultInstance
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/config/AppConfig.scala:99]: JacksonFactory was deprecated. Google says to use this instead

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala [hunk 3]
```diff
@@ -12,8 +12,8 @@ import cats.implicits._
 import com.typesafe.config.ConfigFactory
 import com.typesafe.scalalogging.LazyLogging
 import com.unboundid.ldap.sdk._
-import io.chrisdavenport.log4cats.StructuredLogger
-import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.typelevel.log4cats.StructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala:16]: fallout from version gap

**Comment by MatthewBemis** [comment]
Verified that v1.68 of bouncy castle is pulled in using `sbt dependencyList`

**Review by marctalbott — APPROVED** [review]

**Review by jacmrob — APPROVED** [review]
I appreciate the good commenting in this PR! made it easier to understand

**Pull request merged**